doc/services/binary_descriptors/index.rst

```diff
@@ -0,0 +1,127 @@
+tool.
+
+The binary descriptor header starts with the magic number ``0x2c31b9b9``. It's followed
+by the TLVs, and ends with the ``DESCRIPTORS_END`` (``0xffffffff``) tag. The tags are
+always aligned to 32 bits. If the value of the previous descriptor had a non-aligned
+length, zero padding will be added to ensure that the current tag is aligned.
+
+Putting it all together, here is what the example above would look like in memory
+(of a little endian SoC):
+
+.. code-block::
+
+    b9 b9 31 2c 02 00 00 10 0d 00 00 00 48 65 6c 6c 6f 20 77 6f 72 6c 64 21 00 00 00 00 ff ff ff ff
+   |   magic   |    tag    |  length   | H  e  l  l  o     w  o  r  l  d  !   |   pad  |    end    |
+
+Usage
+*****
+Binary descriptors are always created by the ``DEFINE_BINDESC_*`` macros. As shown in
+the example above, a descriptor can be generated from any string or integer, with any
+ID. However, it is recommended to comply with the standard tags defined in
+``include/zephyr/bindesc.h``, as that would have the following benefits:
+
+ 1. The ``west bindesc`` tool would be able to recognize what the descriptor means and
+    print a meaningful tag
+ 2. It would enforce consistency between various apps from various sources
+ 3. It allows upstream-ability of descriptor generation (see Standard Descriptors)
+
+To define a descriptor with a standard tag, just use the tags included from ``bindesc.h``:
+
+.. code-block:: c
+
+   #include <zephyr/bindesc.h>
+
+   DEFINE_BINDESC_STRING(app_version, BINDESC_TAG_APP_VERSION_STRING, "1.2.3");
+
+Standard Descriptors
+====================
+Some descriptors might be trivial to implement, and could therefore be implemented
+in a standard way in upstream Zephyr. These could then be enabled via Kconfig, instead
+of requiring every user to reimplement them. These include build times, kernel version,
+and host info. For example, to add the build date and time as a string, the following
+configs should be enabled:
+
+.. code-block:: kconfig
+
+   # Enable binary descriptors
+   CONFIG_BINDESC=y
+
+   # Enable definition of binary descriptors
+   CONFIG_BINDESC_DEFINE=y
+
+   # Enable default build time binary descriptors
+   CONFIG_BINDESC_DEFINE_BUILD_TIME=y
+   CONFIG_BINDESC_BUILD_DATE_TIME_STRING=y
+
+For more information read the ``help`` sections of these Kconfig symbols.
+
+west bindesc tool
+=================
+``west`` is able to parse and display binary descriptors from a given executable image.
+It currently supports ``.bin``, ``.hex``, ``.elf`` and ``.uf2`` files. The supported
+subcommands are:
+
+ 1. ``search`` - Prints the value of a specific descriptor found in an image by name
+ 2. ``custom_search`` - Prints the value of a specific descriptor found in an image by type and ID
+ 3. ``dump`` - Prints the values of all descriptors found in an image
+ 4. ``list`` - Prints all known standard descriptor tags
+
+For more information refer to ``west bindesc --help``.
```

Comment:
Please move this under doc/develop/west instead, and add cross-references to that page from here, and to this page from there.